### DISCUSSION TITLE
Hydration completed but contains mismatches.

### DISCUSSION THREAD

**Discussion by ryuhangyeong** [nuxt/nuxt#16042] [Questions] [answered]
In the minimal example below, "Hydration completed but contains mismatches." I am getting an error.

The problem doesn't happen with ssr, but the problem occurs with ssg.

Can you provide any hints to solve the problem?

[nuxt-ssg-dismatch.zip](https://github.com/nuxt/framework/files/10295100/nuxt-ssg-dismatch.zip)

**Comment by TheAlexLichter** [comment, marked as answer]
The problem is that there is a mismatch between HTML coming from server and received on the client.

Check e.g. https://vite-plugin-ssr.com/hydration-mismatch, https://vite-plugin-ssr.com/hydration and feel free to read my [blog post](https://blog.lichter.io/posts/vue-hydration-error/) about it (applies to Nuxt 2 and Nuxt 3).

update: might be https://github.com/nuxt/framework/issues/8051#issuecomment-1272304042

**Reply by ryuhangyeong** [reply]
@manniL Thank you for answer. However, there is no problem when I made HTML in advance with SSG and checked the HTML. However, when I run this from npm run preview, I get a hydration mismatch.

**Comment by ryuhangyeong** [comment]
Even my very minimal example has the same problem. I can't figure out what the heck is going on.

<img width="1193" alt="ssg" src="https://user-images.githubusercontent.com/27342882/209420680-15f51037-71e1-4b8a-a693-7a2e762bb494.png">

```
npm run generate
npm run preview
```

Is the way I tested wrong?

**Reply by TheAlexLichter** [reply]
Could be! Please check https://github.com/nuxt/framework/issues/8051#issuecomment-1272304042

**Reply by ryuhangyeong** [reply]
@manniL thank you!
